packages/web/src/new/scenes/Main/Buy/Buy.tsx

```diff
@@ -16,6 +19,15 @@ const Error = styled.div`
 
 export const Buy: FC = observer(() => {
   const viewModel = useViewModel(BuyViewModel);
+  const location = useLocation<{ fromPage?: string }>();
+
+  useEffect(() => {
+    if (!location.state.fromPage) {
```

Comment:
Are we sure we shouldn't track the event if for some reason there is no `fromPage`?

Reply:
yes. Because this event tracks from what page user came on this page

Reply:
so, if some developer make mistake and don't pass `fromPage` we don't call `track`?

Reply:
I clarified this point with Yura. You were right. This is "page opening" event. Last_Screen parameter completes event info